src/youtubeStream.py

```diff
@@ -0,0 +1,123 @@
+
+    MODEL: ClassVar[Model] = Model(ModelFamily("julie", "camera"), "youtubeStream")
+    REQUIRED_ATTRIBUTES = ["video_url"]
+    
+
+    # Constructor
+    @classmethod
+    def new(cls, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]) -> Self:
+        my_class = cls(config.name)
+        my_class.reconfigure(config, dependencies)
+        return my_class
+    
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.last_call_time = time.time()
+        self.current_frame = 0
+        self.video_cap = None
+
+    # Validates JSON Configuration
+    @classmethod
+    def validate(cls, config: ComponentConfig):
+        # Here we validate config, the following is just an example and should be updated as needed
+        missing_attrs = [attr for attr in cls.REQUIRED_ATTRIBUTES if attr not in config.attributes]
+        if missing_attrs:
+            LOGGER.error("Missing required attributes in Youtube stream Configuration.")
+            raise ValueError(f"Missing required attributes in Youtube stream Configuration: {', '.join(missing_attrs)}")
+        return
+    
+    def reconfigure(self, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]):
+        LOGGER.info("Reconfiguring " + self.name)
+        attrs = struct_to_dict(config.attributes)
+        self.video_url = str(attrs.get("video_url", self.default_url))
+
+        self.config = config
+        self.dependencies = dependencies
+
+        # Release the previous video capture object
+        if self.video_cap:
+            self.video_cap.release()
+
+        # Fetch the actual video URL using yt_dlp
+        stream_url = self.get_video_url(self.video_url)
+
+        # Init. the video capture object with the fetched stream URL
+        self.video_cap = cv2.VideoCapture(stream_url)
+    
+    def get_video_url(self, youtube_url: str) -> str:
+        # Fetch the direct video URL from YouTube using yt_dlp.
+        ydl_opts = {
+            'format': 'best',
+            'quiet': True,
+            'no_warnings': True,
+            'noplaylist': True,
+        }
+        ydl = YoutubeDL(ydl_opts)
+        try:
+            info = ydl.extract_info(youtube_url, download=False)
+            stream_url = info['url']
+            LOGGER.info(f"Fetched video stream URL: {stream_url}")
+            return stream_url
+        except Exception as e:
+            LOGGER.error(f"Error fetching video stream URL: {e}")
+            raise RuntimeError("Failed to fetch the YouTube stream URL.")
+        
+    async def get_image(self, mime_type: str = "", *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs) -> ViamImage:
+        current_time = time.time()
+        
+        for i in range(5):  # Retry getting the video capture up to 10 times
+            ret, frame = self.video_cap.read()
+            if ret:
+                break
+            time.sleep(1)  # Wait 1 second before retrying 
```

Comment:
I think we should change this to `await asyncio.sleep(1)` to not block the entire process